env0/resource_template.go

```diff
@@ -24,32 +24,39 @@ func resourceTemplate() *schema.Resource {
 	}
 }
 
-func templateCreatePayloadRetryOnHelper(d *schema.ResourceData, retryType string, retryOnPtr **client.TemplateRetryOn) {
-	retries, hasRetries := d.GetOk("retries_on_" + retryType)
+func templateCreatePayloadRetryOnHelper(prefix string, d *schema.ResourceData, retryType string, retryOnPtr **client.TemplateRetryOn) {
+	if prefix != "" {
+		prefix += "."
+	}
+	retries, hasRetries := d.GetOk(prefix + "retries_on_" + retryType)
 	if hasRetries {
 		retryOn := &client.TemplateRetryOn{
 			Times: retries.(int),
 		}
-		if retryIfMatchesRegex, ok := d.GetOk("retry_on_" + retryType + "_only_when_matches_regex"); ok {
+		if retryIfMatchesRegex, ok := d.GetOk(prefix + "retry_on_" + retryType + "_only_when_matches_regex"); ok {
 			retryOn.ErrorRegex = retryIfMatchesRegex.(string)
 		}
 
 		*retryOnPtr = retryOn
 	}
 }
 
-func templateCreatePayloadFromParameters(d *schema.ResourceData) (client.TemplateCreatePayload, diag.Diagnostics) {
+func templateCreatePayloadFromParameters(prefix string, d *schema.ResourceData) (client.TemplateCreatePayload, diag.Diagnostics) {
```

Comment:
using "prefix" in case it's the templatless use-case.
Alot of effort/refactoring was done around g, to make these changes as minimal as possible.